lib/maze/plugins/cucumber_report_plugin.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require 'bugsnag'
+require 'cucumber/core/filter'
+require 'json'
+
+# Required to access the options
+module Cucumber
+  class Configuration
+    attr_accessor :options
+  end
+end
+
+module Maze
+  module Plugins
+    class CucumberReportPlugin
+
+      def initialize
+        configured_data = {
+          driver_class: Maze.driver.class,
+          device_farm: Maze.config.farm,
+          device: Maze.config.device,
+          os: Maze.config.os,
+          os_version: Maze.config.os_version
+        }
+        buildkite_data = {
+          pipeline: ENV['BUILDKITE_PIPELINE_NAME'],
+          repo: ENV['BUILDKITE_REPO'],
+          build_url: ENV['BUILDKITE_BUILD_URL'],
+          branch: ENV['BUILDKITE_BRANCH'],
+          message: ENV['BUILDKITE_MESSAGE'],
+          step: ENV['BUILDKITE_LABEL'],
+          commit: ENV['BUILDKITE_COMMIT']
+        }
+        report['configuration'] = configured_data
+        report['build'] = buildkite_data
+      end
+
+      def install_plugin(cuc_config)
+        unless Maze.config.tms_uri && Maze.config.tms_token
+          $logger.warn 'No test report will be delivered for this run'
```

Comment:
Is warn a bit high for people running locally?  Maybe ties in with the above comment.